Tests/FuzzilliTests/WeakSetTest.swift

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import Fuzzilli
+
+class WeakSetTests: XCTestCase {
+
+    func testWeakSetInsert() {
+        var weakSet = WeakSet<TypeExtension>()
+
+        func getTestTypeExtension() -> TypeExtension {
+            return TypeExtension(properties: ["prop"], methods: [], signature: nil)!
+        }
+
+        func addLocalVariable() {
+            let localTypeExtension = getTestTypeExtension()
+            let insertionResult = weakSet.insert(localTypeExtension)
+
+            XCTAssert(weakSet.contains(getTestTypeExtension()))
+            XCTAssert(insertionResult.inserted)
+            XCTAssert(insertionResult.memberAfterInsert === localTypeExtension)
+        }
+        addLocalVariable()
```

Comment:
Could you also test this behaviour without the additional function, by just setting the equivalent of the `localTypeExtension` variable to `nil` or reassigning it, then checking that the set is empty?
Mostly asking because I found the functions name `addLocalVariable` somewhat unintuitive but also don't have a better suggestion for the name =)

Reply:
hmm and isn't it testing different thing? because that test would pass also ordinary `Set` holding pointers.

But I can make it `addLocalVariable(to: inout WeakSet<TypeExtension>)` to be specific

Reply:
Ah yeah that also sounds like a good solution!